src/Surfnet/StepupSelfService/SelfServiceBundle/Security/Authentication/Provider/SamlProvider.php

```diff
@@ -97,4 +108,96 @@ public function supports(TokenInterface $token)
     {
         return $token instanceof SamlToken;
     }
+
+    /**
+     * @param AssertionAdapter $translatedAssertion
+     * @return string
+     */
+    private function getCommonName(AssertionAdapter $translatedAssertion)
+    {
+        $commonNames = $translatedAssertion->getAttributeValue('commonName');
+
+        if (empty($commonNames)) {
+            throw new BadCredentialsException(
+                'No commonName provided'
+            );
+        }
+
+        if (count($commonNames) > 1) {
+            $this->logger->notice('Multiple commonNames provided, picking first one', ['commonNames' => $commonNames]);
```

Comment:
> 1. Is this behaviour accepted by SURF?

I... I... I don't know, my behaviour has generally been pretty well received at SURF...
Heh, but good point, I'll ask @pmeulen 

> 2. We mustn't log PII (common name and e-mail). I think it would be wise to only mention the attribute name.

Ah yes, I heard about that, I'll remove it.